src/main/java/mekanism/common/content/teleporter/TeleporterFrequency.java

```diff
@@ -57,4 +71,28 @@ public Coord4D getClosestCoords(Coord4D coord) {
         }
         return closest;
     }
+
+    @Override
+    protected void read(CompoundNBT nbtTags) {
+        super.read(nbtTags);
+        NBTUtils.setEnumIfPresent(nbtTags, NBTConstants.COLOR, EnumColor::byIndexStatic, (value) -> color = value);
+    }
+
+    @Override
+    protected void read(PacketBuffer dataStream) {
+        super.read(dataStream);
+        color = EnumColor.byIndexStatic(dataStream.readVarInt());
+    }
+
+    @Override
+    public void write(CompoundNBT nbtTags) {
+        super.write(nbtTags);
+        nbtTags.putInt(NBTConstants.COLOR, color.ordinal());
+    }
+
+    @Override
+    public void write(PacketBuffer buffer) {
+        super.write(buffer);
+        buffer.writeVarInt(color.ordinal());
```

Comment:
This can be `buffer.writeEnumValue(color);`

Reply:
ok, should i change it in the qio too?

Reply:
Yes I think that would be a good idea as it makes it more readable overall.

Reply:
done.